packages/netsuite-adapter/src/client/suiteapp_client/suiteapp_file_cabinet.ts

```
@@ -375,13 +373,16 @@ SuiteAppFileCabinetOperations => {
       const subFoldersResults = await querySubFolders(topLevelFoldersResults)
       const foldersResults = topLevelFoldersResults.concat(subFoldersResults)
       const idToFolder = _.keyBy(foldersResults, folder => folder.id)
-      const filteredFolderResults = removeResultsWithoutParentFolder(foldersResults)
-        .map(folder => ({ path: fullPathParts(folder, idToFolder), ...folder }))
+      const [filteredFolderResults, removedFolders] = _.partition(
+        removeResultsWithoutParentFolder(foldersResults)
+          .map(folder => ({ path: fullPathParts(folder, idToFolder), ...folder })),
         // remove excluded folders before creating the query
-        .filter(folder => query.isFileMatch(`${fullPath(folder.path)}${FILE_CABINET_PATH_SEPARATOR}`))
-      const filesResults = await queryFiles(
-        filteredFolderResults.map(folder => folder.id)
+        folder => query.isFileMatch(`${fullPath(folder.path)}${FILE_CABINET_PATH_SEPARATOR}`)
       )
+      log.debug('removed the following %d folder before querying files: %o', removedFolders.length, removedFolders)
+      const filesResults = filteredFolderResults.length > 0 ? await queryFiles(
+        filteredFolderResults.map(folder => folder.id)
+      ) : []
```

Maybe better to throw an error here? otherwise the user will receive a fetch with all the files "deleted" and only in the next fetch they will appear again, as "created"